### PR TITLE
EDI-1642 Improve test coverage

### PR DESCRIPTION
**Pull request by mar-h4**
Added tree structure and utils tests (and some MArray tests) to improve test coverage (currently 99.81%) and to cover possible edge cases.

### CODE REVIEW

File: tests/structure.test.ts [hunk 1]
```diff
@@ -317,4 +432,18 @@ describe('Internal tree structure', () => {
     }
     expect(result_iterate_data).toEqual(result_for_loop);
   });
+
+  it('should find index based on a given condition', () => {
```
Comment (mar-h4) [inline tests/structure.test.ts:436]: I don't know whether this is the expected behavior (this method is not used anywhere yet) - maybe we can improve the documentation @spion ?

**Review by mar-h4 — COMMENTED** [review]
I didn't know how to get to this line https://github.com/hfour/yads/blob/1e52640f845ece33ea7423e234490e9a1041c271/src/tree-structure.ts#L381 - please let me know how to test it since it's the only one left :)

**Comment by spion-h4** [comment]
I think this is as good as it gets using this method. We should probably add property tests comparing the behavior with arrays

File: tests/structure.test.ts (hunk 1 above)
Comment (spion-h4) [reply tests/structure.test.ts:436]: We'll probably be exposing these (similarly to how reduceTo is exposed) when we need to implement monoids.
